config/app/com/microsoft/azure/iotsolutions/uiconfig/services/runtime/ConfigData.java

```diff
@@ -0,0 +1,185 @@
+
+            value = this.keyVault.getKeyVaultSecret(key);
+        }
+
+        return value;
+    }
+
+    @Override
+    public boolean getBool(String key) {
+        Boolean value = false;
+
+        try {
+            value = this.data.getBoolean(key);
+        } catch (ConfigException.Missing e) {
+            // Do Nothing as this goes to KV logic (below)
+        } catch (ConfigException.WrongType e) {
+            // Try to get this as a String and
+            value = this.stringToBoolean(
+                    this.data.getString(key),
+                    null
+            );
+        }
+
+        if (value == null) {
+            String message = String.format("Value for secret %s not found in local env. " +
+                    " Trying to get the secret from KeyVault.", key);
+            log.warn(message);
+
+            value = this.stringToBoolean(
+                    this.keyVault.getKeyVaultSecret(key),
+                    false
+            );
+        }
+
+        return value;
+    }
+
+    @Override
+    public int getInt(String key) throws InvalidConfigurationException {
+        Integer value = null;
+
+        try {
+            value = this.data.getInt(key);
+        } catch (ConfigException.Missing e) {
+            // Do Nothing as this goes to KV logic (below)
+        } catch (ConfigException.WrongType e) {
+            // Try to get this as a String and
+            value = this.stringToInt(
+                    this.data.getString(key),
+                    null
+            );
+        }
+
+        if (value == null) {
+            String message = String.format("Value for secret %s not found in local env. " +
+                    " Trying to get the secret from KeyVault.", key);
+            log.warn(message);
+
+            value = this.stringToInt(
+                    this.keyVault.getKeyVaultSecret(key),
+                    0
+            );
+        }
+
+        return value;
+    }
+
+    public Duration getDuration(String key) {
+        Duration value = null;
+
+        try {
+            value = this.data.getDuration(key);
+        } catch (ConfigException.Missing e) {
+            // Do Nothing as this goes to KV logic (below)
+        } catch (ConfigException.WrongType e) {
+            // Try to get this as a String and
+            value = Duration.of(
+                        Long.valueOf(this.data.getString(key)),
+                        ChronoUnit.SECONDS
+            );
+        }
+
+        if (value == null) {
+            String message = String.format("Value for secret %s not found in local env. " +
+                    " Trying to get the secret from KeyVault.", key);
+            log.warn(message);
+
+            value = Duration.of(
+                    Long.valueOf(this.keyVault.getKeyVaultSecret(key)),
+                    ChronoUnit.SECONDS
+            );
+        }
+
+        return value;
+    }
+
+    @Override
+    public boolean hasPath(String path) {
```

Comment:
Provide some description to what this method does

Reply:
done